usage/app/controllers/UsageApi.scala

```diff
@@ -1,53 +1,121 @@
-import play.api.mvc.Results._
+import com.gu.mediaservice.model.Usage
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import lib._
+import model._
 
 
 object UsageApi extends Controller with ArgoHelpers {
 
-  import Config.{awsCredentials, kahunaUri, keyStoreBucket, loginUriTemplate, rootUri}
+  import Config._
 
   val keyStore = new KeyStore(keyStoreBucket, awsCredentials)
   val Authenticated = auth.Authenticated(keyStore, loginUriTemplate, kahunaUri)
 
+  private def wrapUsage(usage: Usage): EntityReponse[Usage] = {
+    EntityReponse(
+      uri = usageUri(usage.id),
+      data = usage
+    )
+  }
+
+  private def usageUri(usageId: String): Option[URI] = {
+    val encodedUsageId = UriEncoding.encodePathSegment(usageId, "UTF-8")
+    Try { new URI(s"${Config.usageUri}/usages/${encodedUsageId}") }.toOption
+  }
+
   val indexResponse = {
     val indexData = Map("description" -> "This is the Usage Recording service")
     val indexLinks = List(
-      Link("usage", s"$rootUri/usage/media/{id}")
+      Link("usages-by-media", s"${Config.usageUri}/usages/media/{id}"),
+      Link("usages-by-id", s"${Config.usageUri}/usages/{id}")
     )
-    respond(indexData, indexLinks)
+
+    val printPostUri = new URI(s"${Config.usageUri}/usages/print")
+    val actions = List(
+      Action("print-usage", printPostUri, "POST")
+    )
+
+    respond(indexData, indexLinks, actions)
   }
   def index = Authenticated { indexResponse }
 
+  def forUsage(usageId: String) = Authenticated.async {
+    val usageFuture = UsageTable.queryByUsageId(usageId)
+
+    usageFuture.map[play.api.mvc.Result]((mediaUsageOption: Option[MediaUsage]) => {
+      mediaUsageOption.foldLeft(
+        respondNotFound("No usages found.")
+      )((_, mediaUsage: MediaUsage) => {
+        val usage = UsageBuilder.build(mediaUsage)
+        val mediaId = mediaUsage.mediaId
+
+        val uri = usageUri(usage.id)
+        val links = List(
+          Link("media", s"${services.apiBaseUri}/images/${mediaId}"),
+          Link("media-usage", s"${services.usageBaseUri}/usages/media/${mediaId}")
+        )
+
+        respond[Usage](data = usage, uri = uri, links = links)
+      })
+    }).recover { case error: Exception => {
+      Logger.error("UsageApi returned an error.", error)
+      respondError(InternalServerError, "usage-retrieve-failed", error.getMessage())
+    }}
+
+  }
+
   def forMedia(mediaId: String) = Authenticated.async {
     val usagesFuture = UsageTable.queryByImageId(mediaId)
 
-    usagesFuture.map[play.api.mvc.Result](UsageResponseCollection.build).recover { case error: Exception => {
+    usagesFuture.map[play.api.mvc.Result]((mediaUsages: Set[MediaUsage]) => {
+      val usages = mediaUsages.toList.map(UsageBuilder.build)
+
+      usages match {
+        case Nil => respondNotFound("No usages found.")
+        case usage :: _ => {
+          val uri = Try { new URI(s"${services.usageBaseUri}/usages/media/${mediaId}") }.toOption
```

Comment:
`uri` is optional, and `java.net.URI` will throw if it can't parse the string to a URI. So although I don't expect this to throw, I prefer it to `Some(thing)`.